app/models/grant_submission/submission.rb

```diff
@@ -100,5 +109,18 @@ def set_state
     def can_be_unsubmitted?
       errors.add(:base, :reviewed_submission_cannot_be_unsubmitted) if self.reviews.completed.any?
     end
+
+    def prevent_delete_from_published_grant
+      errors.add(:base, :may_not_delete_from_published_grant)
+    end
+
+    def prepare_submission_destroy
```

Comment:
if you keep the dependent destroy on the has_many associations,  such as `has_many :reviews, dependent: :destroy,` and also for criteria_reviews, do you still get the "can't modify frozen hash" error?
we can talk about this more in the POD meeting today if it's hard to explain

Reply:
Instead of "can't modify frozen hash" I got an orphaned records error (if I remember correctly). I ran into it with the "with scored review" spec.

Reply:
yeah probably that error also has to do with the score calculation callbacks. I think this hack is ok. probably unavoidable due to the way the score calculation is done with all those callbacks.